src/jit/compiler.cpp

```diff
@@ -4924,80 +4924,83 @@ int           Compiler::compCompileHelper (CORINFO_MODULE_HANDLE            clas
-        //        
-        if (!compIsForInlining()                       && // We are compiling a method (not inlining one).
-            !hasBeenMarkedAsBadInlinee                 && // The method hasn't been marked as bad inlinee.
-            (opts.eeFlags & CORJIT_FLG_PREJIT)         && // This is NGEN.
-            !forceInline                               && // The size of the method matters.
-            (methodInfo->ILCodeSize > ALWAYS_INLINE_SIZE))
-        {
-            assert(methodInfo->ILCodeSize <= impInlineSize); // Otherwise it must have been marked as a bad inlinee by impCanInlineIL. 
-
-            // We must have run the CodeSeq state machine and got the native size estimate.
-            assert(compNativeSizeEstimate != NATIVE_SIZE_INVALID); 
-
-            int callsiteNativeSizeEstimate = impEstimateCallsiteNativeSize(methodInfo);
-            InlineResult trialResult(this, methodHnd, "prejit2");
-            
-            impCanInlineNative(callsiteNativeSizeEstimate, 
-                               compNativeSizeEstimate,
-                               compInlineeHints,
-                               nullptr, // Calculate static inlining hint.
-                               &trialResult);
+            // Do the initial inline screen.
+            impCanInlineIL(methodHnd, methodInfo, forceInline, &prejitResult);
 
-            if (trialResult.isFailure())
+            // Find the basic blocks. We must do this regardless of
+            // inlineability, since we are prejitting this method.
+            //
+            // Among other things, this will set
+            // compNativeSizeEstimate and compInlineeHints for the
+            // subset of methods we check below.
+            fgFindBasicBlocks();
+
+            // If this method is still a viable inline candidate,
+            // do the profitability screening.
+            if (prejitResult.isCandidate())
+            {
+                // Only needed if the inline is discretionary (not forced)
+                // and the size is over the always threshold.
+                if (!forceInline && (methodInfo->ILCodeSize > ALWAYS_INLINE_SIZE))
+                {
+                    // We should have run the CodeSeq state machine
+                    // and got the native size estimate.
+                    assert(compNativeSizeEstimate != NATIVE_SIZE_INVALID);
+
+                    // Estimate the call site impact
+                    int callsiteNativeSizeEstimate = impEstimateCallsiteNativeSize(methodInfo);
+
+                    // See if we're willing to pay for inlining this method
+                    impCanInlineNative(callsiteNativeSizeEstimate,
+                                       compNativeSizeEstimate,
+                                       compInlineeHints,
+                                       nullptr, // Calculate static inlining hint.
+                                       &prejitResult);
+                }
+            }
```

Comment:
Yeah, that wasn't clear from the source--I had to go look at inline.cpp in order to discover that. Non-blocking; it would only clarify that we can't hit the `else` block of the following `if/else` statement if the condition for this `if` statement is `false`.